core/dbt/adapters/sql/impl.py

```diff
@@ -12,20 +12,35 @@
 from dbt.compat import abstractclassmethod
 
 
+LIST_RELATIONS_MACRO_NAME = 'list_relations_without_caching'
+GET_COLUMNS_IN_RELATION_MACRO_NAME = 'get_columns_in_relation'
+LIST_SCHEMAS_MACRO_NAME = 'list_schemas'
+CHECK_SCHEMA_EXISTS_MACRO_NAME = 'check_schema_exists'
+CREATE_SCHEMA_MACRO_NAME = 'create_schema'
+DROP_SCHEMA_MACRO_NAME = 'drop_schema'
+RENAME_RELATION_MACRO_NAME = 'rename_relation'
+TRUNCATE_RELATION_MACRO_NAME = 'truncate_relation'
+DROP_RELATION_MACRO_NAME = 'drop_relation'
+ALTER_COLUMN_TYPE_MACRO_NAME = 'alter_column_type'
```

Comment:
it occurs to me that these would be a great starting point for building slightly lower level, "unit" tests that could be re-used across adapters. we're missing a standard `create_relation_as` macro. if we had that, you could imagine a test that does, in order:

check if schema exists > drop schema > create schema > create relation as > list relations > rename relation > list relations > list columns in relation > alter column type > list columns in relation > drop relation

if that works, then you can be somewhat confident that you are ready to start building materialization logic. what do you all think? I can open an issue

Reply:
I love this idea. I wish I had it right this moment for my presto adapter.